gson/src/test/java/com/google/gson/stream/JsonReaderTest.java

```diff
@@ -173,7 +173,22 @@ public void testHelloWorld() throws IOException {
     reader.endObject();
     assertEquals(JsonToken.END_DOCUMENT, reader.peek());
   }
-
+  
+  public void testInvalidJsonInput() throws IOException {
+	  String json = "{\n" +
+		        "   \"h\\ello\": true,\n" +
+		        "   \"foo\": [\"world\"]\n" +
+		        "}";
+
+	  try {
+		  JsonReader reader = new JsonReader(reader(json));
```

Comment:
this line and the line following shouldn't be in the try/catch block, it's only the nextName call that throws